clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1546,9 +1546,9 @@ void enqueue(Call call, long now) {
          */
         void call(Call call, long now) {
             if (hardShutdownTimeMs.get() != INVALID_SHUTDOWN_TIME) {
-                log.debug("The AdminClient is not accepting new calls. Timing out {}.", call);
+                log.debug("Cannot accept new call {} when AdminClient is closing.", call);
                 call.handleTimeoutFailure(time.milliseconds(),
```

Comment:
I am thinking if we are throwing an `IllegalStateException`, then we don't need to route it through `handleTimeoutFailure` method. We can directly invoke `call.handleFailure()` from this point and pass the `IllegalStateException` to it. WDYT?

Reply:
That seems like a fine fix. Then we avoid the potentially confusing logging about timeouts that doesn't really apply.

Reply:
Thank you both. @vamossagar12, that makes more sense. I have made an update taking the suggestion.